src/App.tsx

```diff
@@ -1,14 +1,84 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+import { Completed, Filters } from './types/Filters';
 
 export const App: React.FC = () => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filters, setFilters] = useState<Filters>({
+    search: '',
+    completed: Completed.All,
+  });
+
+  const handleClose = () => {
+    setSelectedTodo(null);
+  };
+
+  const handleOpen = (todo: Todo) => {
+    setSelectedTodo(todo);
+  };
+
+  const handleFiltersChange = (
+    key: keyof Filters,
+    value: string | Completed,
+  ) => {
+    setFilters(currentState => ({
+      ...currentState,
+      [key]: value,
+    }));
+  };
+
+  const filteredTodos = useMemo(() => {
+    let filtered = [...todos];
+
+    switch (
+      filters.completed // Completed.All === 'All'
+    ) {
+      case Completed.Active:
+        filtered = filtered.filter(todo => !todo.completed);
+        break;
+
+      case Completed.Completed:
+        filtered = filtered.filter(todo => todo.completed);
+        break;
+      case Completed.All: // break => do nothing
+        break;
+      default:
+        break;
+    }
+
+    // !''
```

Comment:
Remove comments. Try to write obvious code that explains all we need without comments)